### PR TITLE
Use datalim when changing aspect

### LINKED ISSUE #2031: plot_dot
**Describe the bug**
plotdot fig size doesn't behave the way I expect, in that when I set `figsize` in an axes that triple a previous one its not triple the size. There also are some minor bugs where the dots seem to be overlapping some

**To Reproduce**
```
samples = stats.beta(2,2).rvs(100)

width = 10
fig, ax = plt.subplots(figsize=(width, 10))
az.plot_dot(samples, ax=ax)
ax.set_title(f"Width: {width}")
ax.set_xlim(0,1)
```

Then try this, but see that figure is not three times the width
```
width = 30
fig, ax = plt.subplots(figsize=(width, 10))
az.plot_dot(samples, ax=ax)
ax.set_title(f"Width: {width}")
ax.set_xlim(0,1)
```

image
image

**Expected behavior**
Figsize from `plt.subplots` is respected

**Additional context**
Arviz '0.12.0'

### PR DESCRIPTION
**Pull request by ahartikainen**
Fixes #2031 

## Checklist

- [ ] Follows [official](https://github.com/arviz-devs/arviz/blob/main/CONTRIBUTING.md#pull-request-checklist) PR format
- [ ] Includes a sample plot to visually illustrate the changes (only for plot-related functions)
- [ ] New features are properly documented (with an example if appropriate)?
- [ ] Includes new or updated tests to cover the new feature
- [ ] Code style  correct (follows pylint and black guidelines)
- [ ] Changes are listed in [changelog](https://github.com/arviz-devs/arviz/blob/main/CHANGELOG.md#v0xx-unreleased)

### CODE REVIEW

File: arviz/plots/dotplot.py [hunk 1]
```diff
@@ -215,7 +216,11 @@ def wilkinson_algorithm(values, binwidth):
             count += 1
             if count == ndots:
                 break
-        stack_locs.append((stack_first_dot + values[count - 1]) / 2)
+        loc = float((stack_first_dot + values[count - 1]) / 2)
+        if stack_locs and (abs(loc - stack_locs[-1]) < binwidth):
+                stack_locs.append(stack_locs[-1] + binwidth)
+        else:
+            stack_locs.append(loc)
```
Comment (sethaxen) [inline arviz/plots/dotplot.py:223]: Why this particular change?

Comment (ahartikainen) [reply arviz/plots/dotplot.py:223]: If that overlap was indented, then these are not needed

**Comment by ahartikainen** [comment]
@canyon289 does this fix your issue?

Should we test or not

**Comment by canyon289** [comment]
Thanks @ahartikainen. I dont think we need a test unless its easy to do. I can check this out tomorrow (24 hours from now)

**Review by canyon289 — APPROVED** [review]

**Pull request merged**